src/mmw/apps/modeling/mapshed/tasks.py

```diff
@@ -304,6 +313,134 @@ def avg_awc(sjs_result):
     }
 
 
+@shared_task(throws=Exception)
+def nlcd_slope(result):
+    if 'error' in result:
+        raise Exception('[nlcd_slope] {}'.format(result['error']))
+
+    result = parse_sjs_result(result)
+
+    ag_slope_3_count = 0
+    ag_slope_3_8_count = 0
+    ag_count = 0
+    total_count = 0
+
+    for (nlcd_code, slope), count in result.iteritems():
+        if nlcd_code in AG_NLCD_CODES:
+            if slope > 3:
+                ag_slope_3_count += count
+            if 3 < slope < 8:
+                ag_slope_3_8_count += count
+            ag_count += count
+
+        total_count += count
+
+    # percent of AOI that is agricultural with slope > 3%
+    # see Class1.vb#7223
+    ag_slope_3_pct = (float(ag_slope_3_count) / total_count
+                      if total_count > 0 else 0.0)
+
+    # percent of AOI that is agricultural with 3% < slope < 8%
+    ag_slope_3_8_pct = (float(ag_slope_3_8_count) / total_count
+                        if total_count > 0 else 0.0)
+
+    # percent of agricultural parts of AOI with slope > 3%
+    # see Class1.vb#9864
+    n41 = float(ag_slope_3_count) / ag_count if ag_count > 0 else 0.0
+
+    output = {
+        'ag_slope_3_pct': ag_slope_3_pct,
+        'ag_slope_3_8_pct': ag_slope_3_8_pct,
+        'n41': n41,
+    }
+
+    # TODO remove before merging
+    print(output)
+
+    return output
+
+
+@shared_task(throws=Exception)
+def slope(result):
+    if 'error' in result:
+        raise Exception('[slope] {}'.format(result['error']))
+
+    result = parse_sjs_result(result)
+
+    # average slope over the AOI
+    # see Class1.vb#6252
+    avg_slope = result
+
+    output = {
+        'avg_slope': avg_slope
+    }
+
+    # TODO remove before merging
+    print(output)
+
+    return output
+
+
+def get_lu_index(nlcd):
+    if nlcd == 81:
+        lu_index = 1
+    elif nlcd == 82:
+        lu_index = 2
+    elif nlcd in [41, 42, 43]:
+        lu_index = 3
+    elif nlcd in [90, 95]:
+        lu_index = 4
+    elif nlcd in [21, 71]:
+        lu_index = 7
+    elif nlcd in [12, 31]:
+        lu_index = 8
+    elif nlcd == 22:
+        lu_index = 10
+    elif nlcd == 23:
+        lu_index = 11
+    elif nlcd == 24:
+        lu_index = 12
+    elif nlcd == 11:
```

Comment:
Ignore the water value entirely. The last LuCode should be High Density Residential (always 0), not Water.